Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -99,6 +99,7 @@ class ReflectionService
     protected const DATA_PROPERTY_ANNOTATIONS = 15;
     protected const DATA_PROPERTY_VISIBILITY = 24;
     protected const DATA_PROPERTY_TYPE = 26;
+    protected const DATA_PROPERTY_PROMOTED = 26;
```

Comment:
This looks suspicios! Both const have the same value.

Reply:
Upsi 😄 Nice 👌

Reply:
```php
$reflected = new \ReflectionClass($this);
$constants = $reflected->getConstants();
$duplicateConstantValues = array_diff_assoc($constants, array_unique($constants, SORT_NUMERIC));
if ($duplicateConstantValues !== []) {
    throw new \CompileError(sprintf('Class "%s" constants must have unique values. Doubles found: %s', __CLASS__, json_encode($duplicateConstantValues)));
}
```

ich würde ja gerne sowas einbauen als sanity check aber es scheint, dass mehrere konstanten sich einen wert teilen.

```
Class "Neos\Flow\Reflection\ReflectionService" constants must have unique values.
Doubles found: {"DATA_INTERFACE_IMPLEMENTATIONS":1,"DATA_CLASS_INTERFACES":2,"DATA_CLASS_SUBCLASSES":3,"DATA_PARAMETER_SCALAR_DECLARATION":24,"DATA_PARAMETER_ANNOTATIONS":25}
```

und das jetzt anzupassen wäre bestimmt sinvoll, aber man müsste auch die cache clearen ... aber das macht man so oder so ab und zu?

Was sagt ihr @kitsunet @robertlemke?

Reply:
I'd say: Better catch those typos with tests *g